ckanext/switzerland/helpers/validators.py

```diff
@@ -72,55 +85,39 @@ def multilingual_text_output(value):
     return parse_json(value)
 
 
-def date_string_to_timestamp(value):
-    """"
-    Convert a date string (DD.MM.YYYY) into a POSIX timestamp to be stored.
-    Necessary as the date form submits dates in this format.
-    """
-    try:
-        d = parse(str(value), dayfirst=True)
-        epoch = datetime.datetime(1970, 1, 1)
+@register_validator
+def ogdch_date_validator(value):
+    for date_helper in storage_date_helpers:
+        storage_date = date_helper(value)
+        if storage_date:
+            return storage_date
+    log.error("unknown date format detected {}, "
+              "could not be transformed to isodate"
+              .format(value))
 
-        return int((d - epoch).total_seconds())
-    except (TypeError, OverflowError, ParserError):
-        return value
 
+@register_validator
+def ogdch_date_output(value):
+    for date_helper in display_date_helpers:
+        display_value = date_helper(value)
+        if display_value:
+            return display_value
+    log.error("unknown date format detected {}, "
```

Comment:
Same as above. I could investigate this further an see how to deal with empty values.

Reply:
Sorry, can you point me to what you are referring to with "above"?
It could be okay like that, the reason I asked is that I saw the functions that got removed threw an error. Maybe it is not this function's job to validate (the file-name is confusing to me, too, I think, as it seems we have validators but also helpers in here). This function is meant to help displaying a date, so I think that it should have a fallback. Empty is one case but I think it gets even more confusing if a date is of the wrong format (let's hope that we don't have that) but in that case I would rather show a wrong value, so that it is easier to spot and to fix vs hidden. What do you think?